src/server/app/util/services/util.server.service.js

```diff
@@ -116,10 +116,20 @@ module.exports.toLowerCase = function (v){
 };
 
 module.exports.dateParse = function (date) {
-	if (null == date)
+	if (_.isNil(date)) {
 		return null;
+	}
+	if(_.isDate(date)) {
+		return date.getTime();
+	}
+	if(_.isFinite(date)) {
+		return date;
+	}
+	if(_.isString(date)) {
+		return Date.parse(date);
+	}
 
-	return Date.parse(date);
```

Comment:
Yes, I think so. That came to mind later when I was wondering if the BSON objects have some custom .toString method. It'll act the same as before if presented with an object / array / default value. Thanks for the suggestion!